geonode/documents/views.py

```diff
@@ -83,15 +84,23 @@ def document_detail(request, docid):
         if request.user != document.owner:
             Document.objects.filter(id=document.id).update(popular_count=F('popular_count') + 1)
 
+
+        context_dict = {
+            'permissions_json': _perms_info_json(document),
+            'resource': document,
+            'imgtypes': IMGTYPES,
+            'related': related}
+
+        if settings.SOCIAL_ORIGINS:
```

Comment:
Would it be possible to avoid the repetition of this piece of code in the three views?